ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/EthHash.java

```diff
@@ -188,7 +188,7 @@ public static Bytes32 hashHeader(final SealableBlockHeader header) {
     out.writeLongScalar(header.getTimestamp());
     out.writeBytes(header.getExtraData());
     if (header.getBaseFee().isPresent()) {
-      out.writeLongScalar(header.getBaseFee().get());
+      out.writeUInt256Scalar(header.getBaseFee().get());
```

Comment:
same concern here about difference in the write scalar implementation.  I am not sure if this could present a consensus problem or not